[Bug] Searching for a name with a hyphen causes error 500

### Bug Description

Hello. Whenever I search for a string containing a hyphen, Gramps gives a 500 error. For example, an ancestor first name is Pierre-Joseph. If I search for "Pierre" or "Joseph", he shows up in search results. But if I search for "Pierre-Joseph", then Gramps says Error 500.
Logs from Docker:

> 2025-10-13 12:06:57.231 | [2025-10-13 11:06:57 +0000] [17] [ERROR] Error handling request /api/search/?query=pierre-joseph&locale=fr&profile=all&page=1&pagesize=20&type=person
> 2025-10-13 12:06:57.231 | Traceback (most recent call last):
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/gunicorn/workers/sync.py", line 134, in handle
> 2025-10-13 12:06:57.231 |     self.handle_request(listener, req, client, addr)
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/gunicorn/workers/sync.py", line 177, in handle_request
> 2025-10-13 12:06:57.231 |     respiter = self.wsgi(environ, resp.start_response)
> 2025-10-13 12:06:57.231 |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 1536, in __call__
> 2025-10-13 12:06:57.231 |     return self.wsgi_app(environ, start_response)
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 1514, in wsgi_app
> 2025-10-13 12:06:57.231 |     response = self.handle_exception(e)
> 2025-10-13 12:06:57.231 |                ^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 1511, in wsgi_app
> 2025-10-13 12:06:57.231 |     response = self.full_dispatch_request()
> 2025-10-13 12:06:57.231 |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 919, in full_dispatch_request
> 2025-10-13 12:06:57.231 |     rv = self.handle_user_exception(e)
> 2025-10-13 12:06:57.231 |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 917, in full_dispatch_request
> 2025-10-13 12:06:57.231 |     rv = self.dispatch_request()
> 2025-10-13 12:06:57.231 |          ^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/app.py", line 902, in dispatch_request
> 2025-10-13 12:06:57.231 |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/gramps_webapi/api/auth.py", line 44, in wrapper
> 2025-10-13 12:06:57.231 |     return func(*args, **kwargs)
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/views.py", line 110, in view
> 2025-10-13 12:06:57.231 |     return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/flask/views.py", line 191, in dispatch_request
> 2025-10-13 12:06:57.231 |     return current_app.ensure_sync(meth)(**kwargs)  # type: ignore[no-any-return]
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/webargs/core.py", line 650, in wrapper
> 2025-10-13 12:06:57.231 |     return func(*args, **kwargs)
> 2025-10-13 12:06:57.231 |            ^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/gramps_webapi/api/resources/search.py", line 181, in get
> 2025-10-13 12:06:57.231 |     total, hits = searcher.search(
> 2025-10-13 12:06:57.231 |                   ^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/gramps_webapi/api/search/indexer.py", line 319, in search
> 2025-10-13 12:06:57.231 |     results = search.query(
> 2025-10-13 12:06:57.231 |               ^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 |   File "/usr/local/lib/python3.11/dist-packages/sifts/core.py", line 335, in query
> 2025-10-13 12:06:57.231 |     result = conn.execute(fts_query, params) or []
> 2025-10-13 12:06:57.231 |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 2025-10-13 12:06:57.231 | sqlite3.OperationalError: no such column: joseph

### Gramps Web Version Information

Gramps 6.0.4
Gramps Web API 3.3.0
Gramps Web Frontend 25.9.0
Gramps QL 0.4.0
Sifts 1.0.0
locale: en
multi-tree: false
task queue: true
OCR: true
chat: false


### Verification

- [x] I have searched for existing issues before creating this one

--- a/gramps_webapi/_version.py
+++ b/gramps_webapi/_version.py
@@ -21,4 +21,4 @@
 
 """Version information."""
 
-__version__ = "3.4.0"
+__version__ = "3.4.1"

--- a/gramps_webapi/data/apispec.yaml
+++ b/gramps_webapi/data/apispec.yaml
@@ -1,7 +1,7 @@
 swagger: "2.0"
 info:
   title: "Gramps Web API"
-  version: "3.4.0"
+  version: "3.4.1"
   description: >
     The Gramps Web API is a REST API that provides access to family tree databases generated and maintained with Gramps, a popular Open Source genealogical research software package.
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,7 +42,7 @@ dependencies = [
     "pytesseract",
     "gramps-ql>=0.4.0",
     "object-ql>=0.1.3",
-    "sifts>=0.8.3",
+    "sifts>=1.1.0",
     "requests",
     "yclade>=0.5.0",
     "Authlib>=1.6.4",

